include/ccf/crypto/jwk.h

```diff
@@ -88,13 +103,17 @@ namespace crypto
     JsonWebKeyECCurve crv;
     std::string x; // base64url
     std::string y; // base64url
+
+    bool operator==(const JsonWebKeyECPublic&) const = default;
```

Comment:
Just question, what is the reason that we need to define default `==` operator explicitly? Doesn't the compiler create one automatically?

Reply:
I thought that too, but IIRC, it turned out necessary because the `operator==()` was defined for the base object `JsonWebKey` and so comparisons between derived objects (e.g. `JsonWebKeyECPublic`) invoked the base class `operator==()` rather than the derived one.